Created Tests for creating a room and adding a user, and added requirements.txt

kevintumbo opened the pull request:
created tests for creating a room and a person

Commit:
Created Tests for creating a room and adding a user, and added requirements.txt

jimmykimani commented on Tests/test.py:
```diff
@@ -0,0 +1,58 @@
+        self.assertEqual(len(self.dojo.offices), 4, 'offices has been successfully created!')
+
+    def test_can_create_multiple_livingspace(self):
+        self.assertEqual(len(self.dojo.livingspaces), 0)
```
Why are you using country names just asking? they suppose to be names of living spaces

kevintumbo replied:
@jimmykimani It was the first string of names to pop into my head... the outcome in mind was 
`kevin has been allocated living space kenya`

mnjaggah commented on Tests/test.py:
```diff
@@ -0,0 +1,58 @@
+    def test_can_create_office(self):
+        self.assertEqual(len(self.dojo.offices), 0)
+        self.dojo.create_room('office', 'marylin')
+        self.assertEqual(len(self.dojo.offices), 1, 'An office has been successfully created!')
```
assert methods take a msg argument that is the error message when the test fails.
see  https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertEqual
you may need to rephrase to something like 'Unable to create new office'

kevintumbo replied:
@mnjaggah These tests were meant to test for success ... I'm working on tests to test for failure

kevintumbo replied:
@mnjaggah read the documantation. got it. thanks. will apply the changes

Commit:
Created seperate file for testing person(s)

Commit:
created file for person tests

Commit:
Modified dojo.py by adding a destructor

Commit:
"Deleted tests.py"

kevintumbo commented:
made changes on error messages as recomended by @mnjaggah 
added teardown method 
created separate files for room and person tests
removed tests.py

Commit:
modified test messages

Commit:
modified person test functionality

Commit:
modified room test functionality

Commit:
modified room class

Commit:
created person class

Commit:
added functionality thad add person, creates rooms , allocates offices and living space

angiemugo commented on Tests/person_tests.py:
```diff
@@ -0,0 +1,49 @@
+
+    """ Test cases to for successful creation and failed of person(s) """
+
+    def test_can_add_person(self):
```
you have named this test in a way that is misleading.

angiemugo commented on Tests/person_tests.py:
```diff
@@ -0,0 +1,49 @@
+        self.dojo.add_person('John', 'Staff', '')
+        self.dojo.add_person('Anne', 'Fellow', 'Y')
+        self.dojo.add_person('Steve', 'Fellow', 'N')
+        self.assertEqual(len(self.dojo.total_people), 4, 'Person(s) were not added to the system!')
```
maybe this comment should have been persons were added

angiemugo commented on Tests/person_tests.py:
```diff
@@ -0,0 +1,49 @@
+        self.dojo.add_person('', '', 'Y')
+        self.dojo.add_person('', '', '')
+        self.assertEqual(len(self.dojo.total_people), self.num_of_people,
+                         'Person(s) created without necessary credentials')
```
this message is also misleading, it implies that you can add a person without the credentials.

angiemugo commented on Tests/room_tests.py:
```diff
@@ -0,0 +1,55 @@
+        self.dojo = Dojo()
+        self.assertEqual(len(self.dojo.offices), 0)
+        self.dojo.create_room('office', ['marylin'])
+        self.assertEqual(len(self.dojo.offices), 1, 'An office has not been successfully created!')
```
this message is also misleading
i would like you to look at your messages. Apart from tests being used for debugging purposes et al, it also servers as documentation. Your messages should be a documentation of how you expect the code to behave.

angiemugo commented on apps/person.py:
```diff
@@ -0,0 +1,16 @@
+class Person(object):
+    def __init__(self, person_name):
+        self.person_name = person_name
```
What about other attributes of class person i.e person_attributes(fellow, staff)

angiemugo commented on apps/rooms.py:
```diff
@@ -0,0 +1,19 @@
+class Room(object):
+    def __init__(self, room_name):
+        self.room_name = room_name
+        self.occupants = []
```
what about room_type?

Commit:
added class attributes fellow and staff

Commit:
modified test messages

Commit:
Added class attribites staff and fellow

Commit:
added class attributes room typ

Commit:
Modified create room and check availability functionality

Commit:
Added comments to the files

Commit:
added first name and last name arguments, and comments

Commit:
added docopt interface file

mnjaggah commented on dojo.py:
```diff
@@ -0,0 +1,98 @@
+            for room in self.total_rooms:
+
+                if room.type == "office":
+                    if len(room.occupants) >= room.maximum_occupants:
```
Since a room  should not be allocated  more occupants than it's capacity,you could  change this to check if maximum occupancy has been reached i.e. if its equal to and not greater than/equal to

mnjaggah commented on dojo.py:
```diff
@@ -0,0 +1,98 @@
+                            self.available_offices.append(room)
+
+                elif room.type == "living":
+                    if len(room.occupants) >= room.maximum_occupants:
```
Same for the livingspace

mnjaggah reviewed (commented):
Consider fixing inconsistent line spacing especially. between 'if..elif..' statements to comply with PEP8 standards

Commit:
fixed inconsistent line spacing

angiemugo commented on Tests/person_tests.py:
```diff
@@ -0,0 +1,58 @@
+        self.dojo.add_person('', '', 'Y')
+        self.dojo.add_person('', '', '')
+        self.assertEqual(len(self.dojo.total_people), self.num_of_people,
+                         'Person(s) created without necessary credentials')
```
I'd like you to change this message still

kevintumbo replied:
done

Commit:
Modifed person test message

Commit:
added attributes in parent class Room

Commit:
Added attributes to Parent class Room

Commit:
Fixed docopt functionality

Commit:
modified add person method and created room allocation method

Commit:
fixed room allocation bug

Commit:
updated requirements.txt

Commit:
added functionality to put unallocated people in list
